### DISCUSSION TITLE
isDirty case sensitive

### DISCUSSION THREAD

**Discussion by embersaw** [react-hook-form/react-hook-form#9394] [Q&A] [answered]
isDirty property right now is case sensitive.

For example :
You have an input field with initial value : "TEXT".
If you change the input to : "TEXt", the isDirty property becomes true. But if you change it back to "TEXT", isDirty property becomes false. 
Can you please make a change where isDirty property is case insensitive if needed?

**Comment by bluebill1049** [comment]
It would be pretty tricky to cater for such custom logic due to the nature of `isDirty` check is pretty much deep equal with the original `defaultVlaues`, perhaps you can write a single function to asset against.

**Reply by breitembach** [reply]
Do we have a problem with numeric type?

because the isDirty is true when 123 === 123 of type number ?

**Reply by bluebill1049** [reply]
no, i don't think so.

**Comment by leapful** [comment, marked as answer]
@embersaw 

You might need to use a custom hook to get form state, something like: 

```
const useCheckDirty = ({ name, control }) => {
    const values = useWatch({ name, control });
    const { isDirty, dirtyFields, defaultValues } = useFormState({ name, control });

    if (isDirty) {
          // Perform your own dirty check here then return new dirty flag between "values" and "defaultValues" by using key name from "dirtyFields"
   }
   
}
```

Please note this workaround might causes multiple re-rendering because hook `useWatch` is called every time field value is changed (no access to built-in change subscribers from react-hook-form to avoid unnecessary  re-rendering).

Working demo: https://codesandbox.io/s/useformstate-forked-crpnm0?file=/src/App.js
